lib/ReceiptService.php

```diff
@@ -257,6 +257,7 @@ public function createAppStoreReceipt(array $storePurchase)
         $expiresDate = !empty($storePurchase['expires_date']) ? $storePurchase['expires_date'] : null;
         $expiresDatePst = !empty($storePurchase['expires_date_pst']) ? $storePurchase['expires_date_pst'] : null;
         $expiresDateMs = !empty($storePurchase['expires_date_ms']) ? $storePurchase['expires_date_ms'] : null;
+        $promotionalOfferId = !empty($storePurchase['promotional_offer_id']) ? $storePurchase['promotional_offer_id'] : null;
```

Comment:
why don't you do `$storePurchase['promotional_offer_id'] ?? null`

just asking

Reply:
because this library is for php 5, we don't want to break the implementation for all the people using it!